Script to automate the local test procedure (#42)
citycoins/protocol: README.md

### Tests

To be able to fully unit test the functionality contracts with external references (to deployed mainnet contracts)
are copied to a test folder - `tests/contracts/extensions` and the external references replaced by internal testable
target contracts (or stubs). The copy / filter task is automated by running the script here;

```bash
bash scripts/copy-contracts.sh
```

> NB If you are unable to execute the script, run the following command to update the permissions then try again:

Reviewer: What's `NB` in this context?

```bash
chmod 755 scripts/copy-contracts.sh
```

The difference between running the tests with the local, filtered contracts is just the following changes
in Clarinet.toml;

```bash
[contracts.ccd006-city-mining]
path = "tests/contracts/extensions/ccd006-city-mining.clar"

[contracts.ccd007-city-stacking]
path = "tests/contracts/extensions/ccd007-city-stacking.clar"

> ```

Reviewer: Noting this part might change pending convo here: https://github.com/citycoins/protocol/pull/42#issuecomment-1372381619

would be replaced by the lines;

```bash
[contracts.ccd006-city-mining]
path = "contracts/extensions/ccd006-city-mining.clar"

[contracts.ccd007-city-stacking]
path = "contracts/extensions/ccd007-city-stacking.clar"
```

other contracts would follow the same pattern but these are the only two effected at time of writing. The additional tests
which test the local functionality reside in ditrectory;

```bash
tests/extensions/local
```

> and these tests would be expected to fail if the Clarinet.toml was switched back to the default.

Reviewer: No more default/local version :)

#### Automation of Above

> To automate the above strategy;

Reviewer: What if I want to go back to the default config? Do I manually copy the file?

```bash

> scripts/copy-contracts.sh

Reviewer: Shouldn't we specify using bash here since it's a bash script? (that's how I ran it!)

```suggestion
bash scripts/copy-contracts.sh
```

```

> To copy and filter the mining and stacking contracts and copies the `clarinet.local.toml` file to
> `clarinet.toml`. To make the script executable in your environment run

Reviewer: Rewrote this to be a bit clearer:

```suggestion
This script will copy and filter the mining and stacking contracts to create local copies
with local contract references, as well as copy `clarinet.local.toml` to `clarinet.toml`. 
```

> 
> ```bash
> chmod 755 scripts/copy-contracts.sh
> ```

Reviewer: Moved this text to be with the command as a separate item.

```suggestion
If you are unable to execute the script, run the following command to update the permissions then try again:

```bash
chmod 755 scripts/copy-contracts.sh
```
```
